app/src/main/java/com/android/sample/ui/map/Map.kt

```diff
@@ -47,36 +63,45 @@ import com.google.android.gms.maps.CameraUpdateFactory
 import com.google.android.gms.maps.model.BitmapDescriptorFactory
 import com.google.android.gms.maps.model.CameraPosition
 import com.google.android.gms.maps.model.LatLng
-import com.google.maps.android.compose.*
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.Marker
+import com.google.maps.android.compose.MarkerState
+import com.google.maps.android.compose.rememberCameraPositionState
+import com.google.maps.android.compose.rememberMarkerState
 import kotlinx.coroutines.launch
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun MapScreen(
     navigationActions: NavigationActions,
     locationViewModel: LocationViewModel,
-    listActivitiesViewModel: ListActivitiesViewModel
+    listActivitiesViewModel: ListActivitiesViewModel,
 ) {
   val context = LocalContext.current
   val currentLocation by locationViewModel.currentLocation.collectAsState()
   val coroutineScope = rememberCoroutineScope()
   val activities by listActivitiesViewModel.uiState.collectAsState()
+  val activityDetail by listActivitiesViewModel.selectedActivity.collectAsState()
   val defaultLocation = LatLng(46.519962, 6.633597) // EPFL
   var selectedActivity by remember { mutableStateOf<Activity?>(null) }
   var showBottomSheet by remember { mutableStateOf(false) }
+  val previousScreen = navigationActions.getPreviousRoute()
 
-  val firstToDoLocation =
+  val firstLocation =
       try {
-        val loc =
-            (activities as ListActivitiesViewModel.ActivitiesUiState.Success)
-                .activities
-                .firstNotNullOf { it.location }
-        LatLng(loc.latitude, loc.longitude)
+        // the first location that displays is the last activity detail that was checked
+        // if you don't come from an activity detail, it will be centered around user's position
+        if (previousScreen == Screen.ACTIVITY_DETAILS) {
+          activityDetail?.location?.let { LatLng(it.latitude, it.longitude) } ?: defaultLocation
```

Comment:
The best for this would have been a view model, but since our case is simple we can go with this implementation. Note that this works only because we don’t have a bottom bar in the details screen.

Reply:
Yes, this was discussed and approved with the scrum master before. PS: what do you mean with a view model ?